vignettes/hover-label.Rmd

```diff
@@ -0,0 +1,167 @@
+---
+title: "Boxly Hover Label"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Boxly Hover Label}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include=FALSE}
+knitr::opts_chunk$set(
+  echo = TRUE,
+  collapse = TRUE,
+  eval = TRUE,
+  comment = "#>",
+  warning = FALSE,
+  message = FALSE
+)
+```
+
+> This document will guide the developers on how to customize the hover label based on the study needs.
+
+
+# Overview
+
+Hovering is one of the interactive feature for the box plots. When the user hovers the mouse cursor over a specific part of the plot, additional information related to that specific data point or summary statistic is displayed. 
+
+
+Hovering feature in interactive box plots contains two parts:**Hover label of box** and **Hover label of outlier**.
+
+- Hover label of box is to display the descriptive statistics for the plots on the hovering box.
+
+  For example: N, Mean, Median, Q1, Q3, Min and Max
+
+- Hover label of outlier is to display the information for the outlier out of box.
+
+  For example: Subject ID, Change from Baseline and etc.
+  
+  
+# Hover label in function `prepare_boxly()`:
+
+**hover_var_outlier**: A character vector of hover variables for outlier.
+
+```
+prepare_boxly(
+  meta,
+  hover_var_outlier = c("USUBJID", metalite::collect_adam_mapping(meta, analysis)$y)
```

Comment:
Is the `hover_var_outlier` always companied with `hover_outlier_label`, as a 1:1 map? If so, probably we should say something here.

Reply:
Added more details to hover_var_outlier including the one-to-one mapping relationship with hover_outlier_label